target/scripts/startup/setup.d/getmail.sh

```diff
@@ -4,38 +4,59 @@ function _setup_getmail() {
+    local GETMAIL_RC_GENERAL='/etc/getmailrc_general'
 
-    # Generate getmailrc configs, starting with the `/etc/getmailrc_general` base config,
-    # Add a unique `message_log` config, then append users own config to the end.
-    for FILE in /tmp/docker-mailserver/getmail-*.cf; do
-      if [[ -f ${FILE} ]]; then
-        CONFIGS=1
-        ID=$(cut -d '-' -f 3 <<< "${FILE}" | cut -d '.' -f 1)
-        local GETMAIL_CONFIG="${GETMAILRC}/getmailrc-${ID}"
+    # Create the directory /etc/getmailrc.d to place the user config in later.
+    mkdir -p "${GETMAIL_RC_DIR}"
 
-        cat /etc/getmailrc_general                                 >"${GETMAIL_CONFIG}"
-        echo -e "message_log = /var/log/mail/getmail-${ID}.log\n" >>"${GETMAIL_CONFIG}"
-        cat "${FILE}"                                             >>"${GETMAIL_CONFIG}"
-      fi
-    done
+    # Check if getmail config directory exists and at least one <ID>.cf file is present.
+    # getmailrc_general.cf is not mandatory and excluded.
+    if ! find "${GETMAIL_CONFIG_DIR}" -type f -name '*.cf' -not -name getmailrc_general.cf 2>/dev/null | grep -q .; then
+      _log 'warn' 'No getmail configration found'
+      _log 'warn' "Getmail will be disabled"
+    fi
 
-    if [[ ${CONFIGS} -eq 1 ]]; then
-      cat >/etc/cron.d/getmail << EOF
-*/${GETMAIL_POLL} * * * * root /usr/local/bin/getmail-cron
-EOF
-      chmod -R 600 "${GETMAILRC}"
+    # Check if custom getmailrc_general.cf file is present.
+    if [[ -f "${GETMAIL_RC_GENERAL_CF}" ]]; then
+      _log 'debug' "Custom 'getmailrc_general.cf' found"
+      cp "${GETMAIL_RC_GENERAL_CF}" "${GETMAIL_RC_GENERAL}"
     fi
 
-    # Both the debug command and cron job (that runs getmail) for getmail
-    # expect this location to exist.
-    GETMAILDIR=/tmp/docker-mailserver/getmail
-    mkdir -p "${GETMAILDIR}"
+    # If no matching filenames are found, and the shell option nullglob is disabled, the word is left unchanged.
+    # If the nullglob option is set, and no matches are found, the word is removed.
+    shopt -s nullglob
+
+    # Generate getmailrc configs, starting with the `/etc/getmailrc_general` base config, then appending users own config to the end.
+    for FILE in "${GETMAIL_CONFIG_DIR}"/*.cf; do
+      if [[ ${FILE} =~ /getmail/(.+)\.cf && ${FILE} != "${GETMAIL_RC_GENERAL_CF}" ]]; then
+        ID=${BASH_REMATCH[1]}
+
+        _log 'debug' "Processing getmail config '${ID}'"
+
+        GETMAIL_RC=${GETMAIL_RC_DIR}/${ID}
+        cat "${GETMAIL_RC_GENERAL}" "${FILE}" >"${GETMAIL_RC}"
+      fi
+    done
+    chmod -R 600 "${GETMAIL_RC_DIR}"
+
+    # Directory, where "oldmail" files are stored
+    # getmail stores its state - its "memory" of what it has seen in your POP/IMAP account - in the oldmail files.
+    # The debug command for getmail expect this location to exist.
+    GETMAIL_DIR=/var/lib/getmail
+    _log 'debug' "Creating getmail state-dir '${GETMAIL_DIR}'"
+    mkdir -p "${GETMAIL_DIR}"
```

Comment:
Might be better located after the other common local vars, but before the other `mkdir`? I'm not sure if the debug log adds much value, we don't do that for the other `mkdir` 🤔 

```suggestion
    # https://getmail6.org/faq.html#faq-about-oldmail
    # The debug command for getmail expects this location to exist.
    local GETMAIL_STATE_DIR=/var/lib/getmail
    _log 'debug' "Creating getmail state-dir '${GETMAIL_STATE_DIR}'"
    mkdir -p "${GETMAIL_STATE_DIR}"
```

Alternatively:

```suggestion
    # https://getmail6.org/faq.html#faq-about-oldmail
    # The debug command for getmail expects this location to exist.
    mkdir -p /var/lib/getmail
```

Reply:
The variable name `GETMAIL_DIR` comes from the command line option with the same naming: `getmail --getmaildir "${GETMAIL_DIR}" ...`. That's why I haven't named it `GETMAIL_STATE_DIR`.

The reason, for using a variable at all, was to streamline a bit across the various scripts: `debug-getmail`, `getmail-service.sh` and `getmail.sh` have all the same variable defined. This makes the code more familiar between those files.

Regarding the debug log: I can remove it. The main motivation was, the more comments / debug statements, the better 👍

Reply:
Ah ok, fair enough 👍 

I would still go with the comment change (direct link seems better than implicit copy/paste, other line minor typo fix).

The debug log is fine, keeps the context about state dir, I don't know if we log the creation of dir/files as debug or trace level, so whatever makes most sense for that is all good 👍

Reply:
What about:

```
    # Directory, where "oldmail" files are stored: https://getmail6.org/faq.html#faq-about-oldmail
    # The debug command for getmail expect this location to exist.
```

Reply:
Both are fine, except the typo for the last line, `expect` => `expects`.